### DISCUSSION TITLE
24 bit colour

### DISCUSSION THREAD

**Discussion by rowifi1** [Bodmer/TFT_eSPI#881] [Q&A - General] [answered]
Sorry if this is an silly question, can this library work with 24 bit colour values on my Esp32 TTGO Display board?
I'm struggling to make it display properly.

**Comment by Bodmer** [comment, marked as answer]
The library uses either 18 bit, or more usually 16 bit colours (5bits red, 6 bits green, 5 bits blue or "565 colours). These functions are provided for colour conversion:
```
  // Colour conversion
           // Convert 8 bit red, green and blue to 16 bits
  uint16_t color565(uint8_t red, uint8_t green, uint8_t blue);

           // Convert 8 bit colour to 16 bits
  uint16_t color8to16(uint8_t color332);
           // Convert 16 bit colour to 8 bits
  uint8_t  color16to8(uint16_t color565);

           // Convert 16 bit colour to/from 24 bit, R+G+B concatenated into LS 24 bits
  uint32_t color16to24(uint16_t color565);
  uint32_t color24to16(uint32_t color888);
```

**Reply by 9Rune5** [reply]
(googling brought me here -- apologies for reviving an old thread)

What confuses me is that e.g. `TFT_eSPI::drawWideLine()` takes a `uint32_t` color parameter, which eventually ends up calling `pushColor(uint16_t color)`. (unless the GC9A01_DRIVER is defined in which case `drawPixel()` gets called which indeed accepts a `uint32_t color` parameter)

**Reply by Bodmer** [reply]
It is for future developments.
